ignite/engine/engine.py

```diff
@@ -1346,6 +1474,16 @@ def _run_once_on_dataset_legacy(self) -> float:
 
         return time.time() - start_time
 
+    def debug(self, enabled: bool = True) -> None:
```

Comment:
Engine debug mode feature request is described here: https://github.com/pytorch/ignite/issues/1992
Let's revert this revert this change and tackle the feature later in another PR. Thanks!